src/interactive-window/editor-integration/codeGeneratorFactory.ts

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { inject, injectable } from 'inversify';
+import { IDocumentManager } from '../../platform/common/application/types';
+import { IConfigurationService } from '../../platform/common/types';
+import { CodeGenerator } from './codeGenerator';
+import { IGeneratedCodeStorageFactory, IInteractiveWindowCodeGenerator } from './types';
+import { NotebookDocument } from 'vscode';
+
+@injectable()
+export class CodeGeneratorFactory {
+    private readonly cellHashProvidersIndexedByNotebooks = new WeakMap<
```

Comment:
Probably will all happen in clean up, but do want to make sure that we clean out the old cell hash name as much as possible. Love the new name, so I'd want the names in this file to be codeGenerators versus hash providers for the variable names.

Reply:
> old cell hash name as much as possible

Will do that

Reply:
Done